nvtabular/ops/list_slice.py

```diff
@@ -0,0 +1,139 @@
+#
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import cudf
+import cupy as cp
+import numba.cuda
+import numpy as np
+import pandas as pd
+from cudf.core.column import as_column, build_column
+from nvtx import annotate
+
+from nvtabular.dispatch import DataFrameType
+
+from .operator import ColumnNames, Operator
+
+
+class ListSlice(Operator):
+    """Slices a list column
+
+    This operator provides the ability to slice list column by row. For example, to truncate a
+    list column to only include the first 10 elements per row::
+
+        truncated = column_names >> ops.ListSlice(10)
+
+    Take the first 10 items, ignoring the first element::
+
+        truncated = column_names >> ops.ListSlice(1, 11)
+
+    Take the last 10 items from each row::
+
+        truncated = column_names >> ops.ListSlice(-10)
+    """
+
+    def __init__(self, start, end=None):
+        super().__init__()
+        self.start = start
+        self.end = end
+
+        if self.start > 0 and self.end is None:
+            self.end = self.start
+            self.start = 0
+
+        if self.end is None:
+            self.end = np.iinfo(np.int64).max
+
+    @annotate("ListSlice_op", color="darkgreen", domain="nvt_python")
+    def transform(self, columns: ColumnNames, df: DataFrameType) -> DataFrameType:
+        on_cpu = isinstance(df, pd.DataFrame)
+        ret = pd.DataFrame() if on_cpu else cudf.DataFrame()
+        for col in columns:
+            # handle CPU via normal python slicing (not very efficient)
+            if on_cpu:
+                ret[col] = [row[self.start : self.end] for row in df[col]]
+                continue
```

Comment:
What would you think about splitting the CPU and GPU paths into separate clauses of an `if/else`? Purely a style thing, but might be a little easier to read.

Reply:
done!